syntax = "proto3";

message SetAfcTableEntry {
    uint32 app_id = 1;
    uint32 slice_id = 2;
    uint32 packet_id = 3;
    uint32 afc_msg = 4;
}

message PauseFlowRequest {
    repeated SetAfcTableEntry set_afc_table_entries = 1;
}

message PauseFlowResponse {}

enum ScheduleEntryType {
    GeneralUnspecified = 0;
    FixedRangeAdd = 1;
    FixedRangeRemove = 2;
}

message ScheduleEntry {
    ScheduleEntryType type = 1;
    uint32 start = 2;
    uint32 end = 3;
    uint32 target_tor = 4;
    optional uint32 next_hop = 5;
}

message ResumeFlowRequest {
    repeated ScheduleEntry schedule_entries = 1;
}

message ResumeFlowResponse {}

message ClearRequest {}

message ClearResponse {}

service Rpc {

rpc PauseFlow(PauseFlowRequest) returns (PauseFlowResponse) {}

rpc ResumeFlow(ResumeFlowRequest) returns (ResumeFlowResponse) {}

rpc Clear(ClearRequest) returns (ClearResponse) {}

}
